From d77b46810ebb3cca096d941430bf86eaa0ba2e9d Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Fri, 19 Jun 2020 13:26:05 +0200
Subject: [PATCH 01/10] profiles: add the wlan-ap board profiles

These are the profiles describing the boards and feature sets that we need
to build wlan-ap packages.

Signed-off-by: John Crispin <john@phrozen.org>
---
 profiles/ea8300.yml  |  5 +++
 profiles/ec420.yml   |  5 +++
 profiles/ecw5211.yml |  5 +++
 profiles/ecw5410.yml |  5 +++
 profiles/wlan-ap.yml | 81 ++++++++++++++++++++++++++++++++++++++++++++
 5 files changed, 101 insertions(+)
 create mode 100644 profiles/ea8300.yml
 create mode 100644 profiles/ec420.yml
 create mode 100644 profiles/ecw5211.yml
 create mode 100644 profiles/ecw5410.yml
 create mode 100644 profiles/wlan-ap.yml

diff --git a/profiles/ea8300.yml b/profiles/ea8300.yml
new file mode 100644
index 0000000000..cb5f4b25fe
--- /dev/null
+++ b/profiles/ea8300.yml
@@ -0,0 +1,5 @@
+---
+profile: linksys_ea8300
+target: ipq40xx
+subtarget: generic
+description: Build image for the Linksys EA8300
diff --git a/profiles/ec420.yml b/profiles/ec420.yml
new file mode 100644
index 0000000000..7cc30b883f
--- /dev/null
+++ b/profiles/ec420.yml
@@ -0,0 +1,5 @@
+---
+profile: tp-link_ec420_g1
+target: ipq40xx
+subtarget: generic
+description: Build image for the TP-Link EC420 G1
diff --git a/profiles/ecw5211.yml b/profiles/ecw5211.yml
new file mode 100644
index 0000000000..9150a16126
--- /dev/null
+++ b/profiles/ecw5211.yml
@@ -0,0 +1,5 @@
+---
+profile: edgecore_ecw5211
+target: ipq40xx
+subtarget: generic
+description: Build image for the Edgecore ECW5211
diff --git a/profiles/ecw5410.yml b/profiles/ecw5410.yml
new file mode 100644
index 0000000000..d7bfba8560
--- /dev/null
+++ b/profiles/ecw5410.yml
@@ -0,0 +1,5 @@
+---
+profile: edgecore_ecw5410
+target: ipq806x
+subtarget: generic
+description: Build image for the Edgecore ECW5410
diff --git a/profiles/wlan-ap.yml b/profiles/wlan-ap.yml
new file mode 100644
index 0000000000..b5453c1151
--- /dev/null
+++ b/profiles/wlan-ap.yml
@@ -0,0 +1,81 @@
+---
+description: Add the wlan-ap dependencies
+feeds:
+  - name: wlan_ap
+    path: ../../feeds/wlan-ap
+
+packages:
+  - cgi-io
+  - coreutils
+  - coreutils-sleep
+  - jansson
+  - kmod-crypto-crc32c
+  - kmod-crypto-hash
+  - kmod-lib-crc32c
+  - kmod-mpls
+  - kmod-nf-nat6
+  - kmod-openvswitch
+  - libatomic
+  - libcares
+  - libev
+  - libiwinfo-lua
+  - liblua
+  - liblucihttp
+  - liblucihttp-lua
+  - libmosquitto-ssl
+  - libopenssl
+  - libpcap
+  - libprotobuf-c
+  - librt
+  - libstdcpp
+  - libubus-lua
+  - libunbound-light
+  - libuuid
+  - lua
+  - luci
+  - luci-app-firewall
+  - luci-app-opkg
+  - luci-base
+  - luci-lib-ip
+  - luci-lib-jsonc
+  - luci-lib-nixio
+  - luci-mod-admin-full
+  - luci-mod-network
+  - luci-mod-status
+  - luci-mod-system
+  - luci-proto-ipv6
+  - luci-proto-ppp
+  - luci-theme-bootstrap
+  - openvswitch
+  - openvswitch-common
+  - openvswitch-libofproto
+  - openvswitch-libopenvswitch
+  - openvswitch-libovsdb
+  - openvswitch-ovsdb
+  - openvswitch-vswitchd
+  - protobuf
+  - protobuf-lite
+  - rpcd
+  - rpcd-mod-file
+  - rpcd-mod-iwinfo
+  - rpcd-mod-luci
+  - rpcd-mod-rrdns
+  - uhttpd
+  - uuidgen
+  - zlib
+  - opensync
+  - wpad-openssl
+  - ip-bridge
+
+diffconfig: |
+  CONFIG_OPENSSL_ENGINE=y
+  CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM=y
+  CONFIG_OPENSSL_WITH_ASM=y
+  CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
+  CONFIG_OPENSSL_WITH_CMS=y
+  CONFIG_OPENSSL_WITH_DEPRECATED=y
+  CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
+  CONFIG_OPENSSL_WITH_PSK=y
+  CONFIG_OPENSSL_WITH_SRP=y
+  CONFIG_OPENSSL_WITH_TLS13=y
+  # CONFIG_PACKAGE_wpad-basic is not set
-- 
2.25.1

